plugin/effect/src/index.ts

```diff
@@ -0,0 +1,38 @@
+import { Effect } from "effect";
+import * as z from "zod";
+
+function zodEffect(this: z.ZodType, data: unknown) {
+  return Effect.tryPromise({
+    try: async () => this.parseAsync(data),
+    catch(error) {
+      return error as z.ZodError; // only kinda a lie...
+    },
+  });
+}
+
+function zodEffectSync(this: z.ZodType, data: unknown) {
+  return Effect.try({
+    try: () => this.parse(data),
+    catch(error) {
+      return error as z.ZodError; // only kinda a lie...
+    },
+  });
+}
+
+let executed = false;
+if (!executed) {
+  executed = true;
+  z.ZodType.prototype.effect = zodEffect;
+  z.ZodType.prototype.effectSync = zodEffectSync;
+}
```

Comment:
Not sure you need to have an `executed` check. 
imports are only run once, so modules are already singletons.
But maybe i'm missing something.

Reply:
This gets compiled to CJS as well and I'm not sure the same guarantee exists.